### PR TITLE
Add search suggestion

### PR DESCRIPTION
**Pull request by marians**
Towards https://github.com/giantswarm/giantswarm/issues/14703

This PR adds a "Did you mean" suggestion feature to search.

Requires https://github.com/giantswarm/docs-indexer/pull/33

The suggestions will only appear if the term(s) in the suggestion are more frequent than the ones entered by the user.

## Previews

image

image

### CODE REVIEW

File: src/assets/scripts/base.js [hunk 1]
```diff
@@ -86,27 +107,42 @@ function doSearch(q) {
       console.warn("Error in ajax call to /searchapi/:", textStatus, errorThrown);
     },
     success: function(data){
-      $(".result").empty();
+      // Display suggestions
+      var hasSuggestion = false;
+      if (data.suggest["phrase-suggester"] && data.suggest["phrase-suggester"].length > 0 && data.suggest["phrase-suggester"][0].options.length > 0) {
```
Comment (axbarsan) [inline src/assets/scripts/base.js:112]: How about storing `data.suggest["phrase-suggester"]` into a variable?

File: src/assets/scripts/base.js [hunk 2]
```diff
@@ -86,27 +107,42 @@ function doSearch(q) {
...
+        $(".suggestion").find("a").attr("href", "./?q=" + encodeURI(data.suggest["phrase-suggester"][0].options[0].text)).text(data.suggest["phrase-suggester"][0].options[0].text);
+        $(".suggestion").show();
```
Comment (axbarsan) [inline src/assets/scripts/base.js:114]: Same with `$(".suggestion")`?

File: src/assets/scripts/base.js (hunk 1 above)
Comment (marians) [reply src/assets/scripts/base.js]: Yep, done

File: src/assets/scripts/base.js (hunk 2 above)
Comment (marians) [reply src/assets/scripts/base.js]: Done

**Review by axbarsan — APPROVED** [review]

**Pull request merged**